tests/testthat/test-verb-filter.R

```diff
@@ -46,6 +62,61 @@ test_that(".preserve is not supported", {
   expect_snapshot(error = TRUE, lf %>% filter(x == 1, .preserve = TRUE))
 })
 
+test_that("filter() works with mutate()", {
+  lf <- lazy_frame(x = 1, y = 2)
+  named_list <- function(...) {
+    out <- list(...)
+    if (is_null(names(out))) {
+      names(out) <- names2(out)
+    }
```

Comment:
What's this doing/why is it needed? I'm guessing it's something to do with `names(x)` being `NULL` vs `""`?